src/Buffer/FSBuffer.cpp

```diff
@@ -9,22 +9,26 @@ FSBuffer::~FSBuffer(){
 }
 
 size_t FSBuffer::available(){
-	return size - cursor;
+	return bytesFilled - cursor;
 }
 
 bool FSBuffer::moveRead(size_t amount){
-	if(cursor + amount > size) return false;
+	if(cursor + amount > bytesFilled) return false;
 	cursor += amount;
 	return true;
 }
 
-void FSBuffer::refill(){
-	if(cursor != size){
-		memmove(buffer, buffer + cursor, size - cursor);
+bool FSBuffer::refill(){
+	if(cursor != bytesFilled){
+		memmove(buffer, buffer + cursor, max(int(bytesFilled - cursor), 0));
```

Comment:
za max ti oba parametra trebaju biti isti tip podatka. u ovom slučaju odlučio si iz size_t pretvorit u int da bi bio isti tip kao i 0, ali size_t može držat duplo veće brojeve od int. ako se desi da je bytesFilles - cursor veći broj nego int može držat, to će se na kraju wrappat i bit će neki drugi broj različitog predznaka, a budući da tu s max ograničavaš donju granicu, ako se desi wrap će ti vratit neki yuge broj i memmove će ti crashat sve.
bolje castaj ovu nulu u size_t